src/components/ListingCard/index.tsx

```diff
@@ -25,20 +25,42 @@ export default function ListingCard({
       onClick={onClick ? () => onClick(caseData.id) : undefined}
     >
       {/* hard-coded for now */}
-      <H2>Case title.</H2>
-      <p>
-        <strong>Languages: </strong>
-        {caseData.languages.join(', ')}
-      </p>
-      <p>
-        <strong>Upcoming Date: </strong>
-        {parseDate(timestampStringToDate(caseData.upcoming_date))}
-      </p>
+      <CardTitle>
+        Guatemalan mother and two children seeking protection
+      </CardTitle>
+
       <TagRow>
-        {caseData.needs_interpreter && (
-          <CardTag color={COLORS.interpreterColor}>Interpreter needed</CardTag>
+        {caseData.needs_interpreter && caseData.needs_attorney && (
+          <CardTag color={COLORS.tagColor}>
+            Needs Interpreter & Attorney
+          </CardTag>
+        )}
+        {caseData.needs_interpreter && !caseData.needs_attorney && (
+          <CardTag color={COLORS.tagColor}>Needs Interpreter</CardTag>
+        )}
+        {!caseData.needs_interpreter && !caseData.needs_attorney && (
+          <CardTag color={COLORS.tagColor}>Needs Attorney</CardTag>
+        )}
+        {caseData.is_remote ? (
+          <CardTag color={COLORS.tagColor}>Remote</CardTag>
+        ) : (
+          <CardTag color={COLORS.tagColor}>In Person</CardTag>
+        )}
+        {caseData.in_court ? (
+          <CardTag color={COLORS.tagColor}>Court</CardTag>
+        ) : (
+          <CardTag color={COLORS.tagColor}>USCIS</CardTag>
+        )}
+        {caseData.languages && (
+          <CardTag color={COLORS.tagColor}>
+            {caseData.languages.join(', ')}
+          </CardTag>
         )}
       </TagRow>
+      <P>
+        <strong>Next Filing/Court Date: </strong>
+        {parseDate(timestampStringToDate(caseData.upcoming_date))}
```

Comment:
we want a null check here! something like
`caseData.upcoming_date ? parseDate(timestampStringToDate(caseData.upcoming_date)) : 'N/A'`?

Reply:
solved!